app/Listeners/QueueMessage.php

```diff
@@ -35,16 +37,18 @@ public function handle(QueueMessageRequest $event)
         // Build the email.
         $email = new Email($resources);
 
-        foreach ($email->allMessages as $content) {
-            $settings = [
-                'subject' => $content['message']['subject'],
-                'from' => $email->contest->sender_email,
-                'from_name' => $email->contest->sender_name,
-                'to' => $content['user']->email,
-                'to_name' => $content['user']->first_name,
-            ];
+        if (isset($email->allMessages)) {
+            foreach ($email->allMessages as $content) {
+                $settings = [
+                    'subject' => $content['message']['subject'],
+                    'from' => $email->contest->sender_email,
+                    'from_name' => $email->contest->sender_name,
+                    'to' => $content['user']->email,
+                    'to_name' => $content['user']->first_name,
+                ];
 
-            $this->sendMail($content['message'], $settings);
+                $this->sendMail($content['message'], $settings);
```

Comment:
The `Email` class takes in an array of users to send messages to. In the case of a welcome email it passed an array of only one user. The code you point to, goes through each user passed, checks if they have valid email address and then adds it to an `allMessages` property that we loop through here. 

The issue that I am trying to fix here is when that one user passed, doesn't have a valid email, so no `allMessage` property is passed and we don't want to try to loop through it. 

So welcome emails will still send, when the user is valid, cause we will build a message for them. I added a line [to log](https://github.com/DoSomething/gladiator/pull/294/files#diff-4f39f258033e5d77d4d966ae87da96c2R167) when a user doesn't pass validation so we can see the user that is failing.